specification/containerservice/resource-manager/Microsoft.ContainerService/stable/2021-07-01/managedClusters.json

```diff
@@ -2371,6 +2371,9 @@
           "type": "boolean",
           "description": "Whether to enable auto-scaler"
         },
+        "scaleDownMode": {
+          "$ref": "#/definitions/ScaleDownMode"
```

Comment:
They flag this as missing a `title` or `description`. It's _technically_ not (see https://github.com/Azure/azure-rest-api-specs/issues/14699) but with that said I think our pattern of doing this is actually sorta wrong.

There are two things:
1. The property (here).
2. The type.

The two shouldn't always have the same description. In your case, if somebody else comes along in a new API version and used this enum for something else it might not have _exactly_ the same semantics. For example maybe their usage doesn't impact cluster autoscaler. Or maybe it would default differently.

In order to play nicely with their linter (and also to distinguish between the type and its usage better than we do) I wonder if it'd be good to put the following here:
```
"title": "The scale down mode to use when scaling the Agent Pool.",
"description": "This also effects the cluster autoscaler behavior. If not specified, the default is ???"
```

You can also optionally actually use the `default` field in Swagger although I observe we don't use it all that often and I'm not too picky either way (as long as we somehow say what the default is, either in the comment or via the annotation - which will become a comment in all the languages that I'm aware of. IIRC a "The default is <default>" gets tacked onto the end if you actually use the `default` annotation)

Reply:
very interesting points here. I go over the Batch definitions and it does seem there's a fair chunk of improvements we can make specifically regarding that point with property and types.